webapp/src/pages/Settings.tsx

```diff
@@ -194,11 +201,14 @@ const NumberField: React.FC<
 
 const Settings: React.FC = () => {
   const { jobId } = useParams<{ jobId: string }>();
+  const [language, setLanguage] = React.useState<
+    SupportedLanguage | undefined
+  >();
   const {
     data: defaultConfig,
     isLoading,
     error,
-  } = getDefaultConfigEndpoint.useQuery({ jobId });
+  } = getDefaultConfigEndpoint.useQuery({ jobId, language });
```

Comment:
With this setup, the default config is initially fetched with `language: undefined`, which defaults to English in the back end. Let's imagine a scenario where
- the `config.language` is French;
- the `config.behavioral_testing` is `null`;
- the user doesn't touch the language selector, so the `defaultConfig` stays English;
- the user enables `behavioral_testing`; then
- the `partialConfig` will be populated with the English default value, even though `config.language` is French.

I think the solution is to fetch `getDefaultConfigEndpoint` after `getConfigEndpoint` (so moving it down), using `language: language ?? resultingConfig.language` so it's never `undefined`.
```suggestion
  } = getDefaultConfigEndpoint.useQuery({
    jobId,
    language: language ?? resultingConfig.language,
  });
```

Reply:
Yes!. True.. Done.